clang/lib/3C/AVarBoundsInfo.cpp

```diff
@@ -470,11 +470,11 @@ bool AvarBoundsInference::inferBounds(BoundsKey K, AVarGraph &BKGraph, bool From
 
 bool AVarBoundsInfo::isValidBoundVariable(clang::Decl *D) {
   // All parameters, return, and field values are valid bound variables.
-  if (isa<ParmVarDecl>(D) || isa<FunctionDecl>(D) || isa<FieldDecl>(D))
+  if (D && (isa<ParmVarDecl>(D) || isa<FunctionDecl>(D) || isa<FieldDecl>(D)))
     return true;
  
   // For VarDecls, check if these are are not dummy and have a name.
-  if (auto *VD = dyn_cast<VarDecl>(D))
+  if (auto *VD = dyn_cast_or_null<VarDecl>(D))
```

Comment:
Why is it that `D` could now be `nullptr` here when it couldn't before? Or maybe it could, and you are fixing a bug? When could that happen?

Reply:
I started explicitly passing null when needing a param for a prototype that doesn't exist i.e `int *foo()`. This code is called from that section, so it needed null checks. Previously, it may have been worthy of an assert.